### ISSUE TITLE
Rule Request: redundant logfmt str ctor catch in G rules or add new rule

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#12734]
This is a really common antipattern I wish logfmt (G*) rules would catch:
```python
   log.warning("Cannot write to %s", str(profile_path))
```

is redundant and does an potentially expensive str construction which is what we are trying to avoid with these rules anyway. Instead, the `%s` ensure it will be formatted as a string anyway! As such, this can be simplified into the better code:
```python
   log.warning("Cannot write to %s", profile_path)
```
Which will be faster in both the case where it's printed and the case where it's not by either avoiding an unnecessary string copy or a unnecessary construction in the first place.

This rule could be added more generically to any string formatting in the new or old styles

**Comment by adamchainz** [comment]
Replying on behalf of flake8-logging here. I think we can avoid parsing the string formatting the message and instead error for any `str` or `repr` calls in the positional arguments of a logging call. That seems like it would catch the issue without much chance of false positive. What do you think? Happy to review a PR there.

**Comment by charliermarsh** [comment]
Thanks! We do have a parser for format strings so we _could_ parse it and cross-reference with the positional calls if needed.
